### ISSUE TITLE
BHoM_Adapter: async push and pull?

### ISSUE THREAD

**Issue by epignatelli** [BHoM/BHoM_Adapter#50]
An open question more than a issue.

It happens some times that when dealing with large sets of data the push/pull components take long time.

Would it be beneficial to switch to async methods?

**Comment by alelom** [comment]
Worth doing it in 3.1

**Comment by IsakNaslundBh** [comment]
Would be a bit careful with this in an environment like GH where a user could be dragging sliders back and forward. Afraid this could lead to a lot of weird/unexpected behaviours.

**Comment by adecler** [comment]
Good point @IsakNaslundBh . Although, we are already facing a similar problem with toolkits like Socket_Toolkit where things are already asynchronous. So a solution for that one could also be applied to any toolkit that supports async push and pulls. First things that comes to mind is a queue system where every intermediate request in the queue is skipped to only cover the last one.

**Comment by alelom** [comment]
Good points raised. Since there's already enough on the plate for the Refactoring for the 3.0, will postpone further planning/discussion on this for later on.
